### PR TITLE
GUVNOR-3547: [Guided Decision Table] Patterns from BRL conditions are not available for another action columns

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3547

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:

- https://github.com/kiegroup/drools/pull/1529
- https://github.com/kiegroup/drools-wb/pull/648

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/BRLRuleModel.java [hunk 1]
```diff
@@ -256,14 +256,19 @@ public boolean isBoundFactUsed(final String binding) {
                 if (action.getBoundName().equals(binding)) {
                     return true;
                 }
+            } else if (col instanceof ActionSetFieldCol52) {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/BRLRuleModel.java:259]: Here's the change for ```drools-workbench-models``` - check if a binding is used by a ```ActionSetFieldCol52``` too

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/BRLRuleModel.java [hunk 2]
```diff
@@ -256,14 +256,19 @@ public boolean isBoundFactUsed(final String binding) {
...
+                }
             } else if (col instanceof ActionRetractFactCol52) {
 
                 if (col instanceof LimitedEntryActionRetractFactCol52) {
 
                     //Check whether Limited Entry retraction is bound to Pattern
                     final LimitedEntryActionRetractFactCol52 ler = (LimitedEntryActionRetractFactCol52) col;
                     if (ler.getValue().getStringValue().equals(binding)) {
-                        return false;
+                        return true;
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/main/java/org/drools/workbench/models/guided/dtable/shared/model/BRLRuleModel.java:271]: I also noticed, when writing Unit Tests, that this was wrong for ```LimitedEntryActionRetractFactCol52```

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 3]
```diff
@@ -1647,19 +1659,132 @@ public void testRHSDelimitedNonEmptyStringValues() {
...
                 "end";
 
-        assertEqualsIgnoreWhitespace( expected4,
-                                      drl4 );
+        assertEqualsIgnoreWhitespace(expected4,
+                                     drl4);
+    }
+
+    @Test
+    public void testRuleModelWithRHSBoundFactsUsageWithActionSetField() {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1667]: New test..

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 4]
```diff
@@ -1647,19 +1659,132 @@ public void testRHSDelimitedNonEmptyStringValues() {
...
+
+        final BRLRuleModel model = new BRLRuleModel(dt);
+
+        //Checks
+        assertTrue(model.isBoundFactUsed("$d"));
+        assertFalse(model.isBoundFactUsed("$cheese"));
+    }
+
+    @Test
+    public void testRuleModelWithRHSBoundFactsUsageWithActionInsertFact() {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1689]: New test..

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 5]
```diff
@@ -1647,19 +1659,132 @@ public void testRHSDelimitedNonEmptyStringValues() {
...
+
+        final BRLRuleModel model = new BRLRuleModel(dt);
+
+        //Checks
+        assertTrue(model.isBoundFactUsed("$ins"));
+        assertFalse(model.isBoundFactUsed("$cheese"));
+    }
+
+    @Test
+    public void testRuleModelWithRHSBoundFactsUsageWithActionRetractFact() {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1713]: New test..

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 6]
```diff
@@ -1647,19 +1659,132 @@ public void testRHSDelimitedNonEmptyStringValues() {
...
     }
 
-    private void assertEqualsIgnoreWhitespace( final String expected,
-                                               final String actual ) {
-        final String cleanExpected = expected.replaceAll( "\\s+",
-                                                          "" );
-        final String cleanActual = actual.replaceAll( "\\s+",
-                                                      "" );
+    @Test
+    public void testRuleModelWithRHSBoundFactsUsageWithLimitedEntryActionRetractFact() {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1737]: New test..

File: drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java [hunk 7]
```diff
@@ -1647,19 +1659,132 @@ public void testRHSDelimitedNonEmptyStringValues() {
...
+
+        final BRLRuleModel model = new BRLRuleModel(dt);
+
+        //Checks
+        assertTrue(model.isBoundFactUsed("$d"));
+        assertFalse(model.isBoundFactUsed("$cheese"));
     }
 
+    @Test
+    public void testRuleModelWithRHSBoundFactsUsageWithBRLActionColumn() {
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-guided-dtable/src/test/java/org/drools/workbench/models/guided/dtable/backend/BRLRuleModelTest.java:1758]: New test..

**Review by karreiro — APPROVED** [review]
Thank you, @manstis.

**Review by jomarko — APPROVED** [review]
Looks good to me, still waiting for full downstream build to approve also to second on of the ensemble.

**Pull request merged**